presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -2002,7 +2002,16 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
 
         Optional<HiveBucketHandle> hiveBucketHandle = getHiveBucketHandle(table, typeManager);
         if (!hiveBucketHandle.isPresent()) {
-            return Optional.empty();
+            // return preferred layout which is partitioned by partition columns
```

Comment:
Apply partitioning by partition columns also when the table is bucketed.

Reply:
When table is bucketed then data partitioning on partition columns should be required, but this is outside of scope of this PR.
For example,
* currently: if you have 10 buckets per partition, then we partition data by bucket columns. This means only 10 workers will write data for all partitions
* better: if you partition data by bucket columns AND partition columns, then still 10 workers will write data per partition. However, set of workers per partition changes therefore improving parallelism.